Not possible to specify a custom version via the interactive UI

<!--- Provide a short summary of the issue in the title above -->

## Description

Hi 👋 

First off: thanks a ton for your work on this really awesome tool 🦄 💜.

### Problem

I'm currently running into a problem when trying to a problem when trying to specify a custom version _(like `v1.0.0.beta`)_ via the interactive UI: Selecting `Other (specify)` does not show the successive prompt to input the version number string. The release process then fails with an error, since the version number is blank.

The explicit error varies for `npm` and `yarn`, but the core problem is the same: they are missing the version number to update to. Here two examples of the error messages:

**yarn**
<details>
<summary>click here to expand example</summary>

```
✖ Command failed with exit code 1: yarn version --new-version
error Invalid version supplied.
info Current version: 3.0.0
info Visit https://yarnpkg.com/en/docs/cli/version for documentation about this command.
error Command failed with exit code 1.
```
</details>

**npm**
<details>
<summary>click here to expand example</summary>
```
✖ Command failed with exit code 1: npm version
npm ERR! npm version [<newversion> | major | minor | patch | premajor | preminor | prepatch | prerelease [--preid=<prerelease-id>] | from-git]
npm ERR! (run in package dir)
npm ERR! 'npm -v' or 'npm --version' to print npm version (6.13.7)
npm ERR! 'npm view <pkg> version' to view a package's published version
npm ERR! 'npm ls' to inspect current package/dependency versions
error Command failed with exit code 1.
```
</details>

### Cause

The interactive UI skips showing the [next input](https://github.com/sindresorhus/np/blob/f72c30e65fb40822ee1b1191c227816b3334a374/source/ui.js#L86) even though the `version` is not specified yet and instead continues with the next step.

The next step fails due to the `version` still being blank, which according to [this line](https://github.com/sindresorhus/np/blob/f72c30e65fb40822ee1b1191c227816b3334a374/source/ui.js#L90) would normally have been the condition to actually show the prompt for inputting a custom version 🤔.

### Scenarios

I run into this scenario regardless of using `npm` or `yarn`, `zsh` or `bash` and regardless of "how" I execute `np`:
1. in a `release` script command defned in my `package.json`
1. directly calling `yarn np` or `node_modules/.bin/np`
1. from a shell script

The behavior is the same for versions `v6.2.3`, `v6.0.0` and `v5.1.1` _(I did not try other versions as I felt the problem is probably somewhere on my side)_.

### Question

Am I using the tool wrong? I need to produce versions with names like e.g. `v3.2.0beta` and would ideally want to do this via the interactive UI 🤔.

## Steps to reproduce

Here are the steps for a minimal example to reproduce this problem _(also available as [gist](https://gist.github.com/klappradla/084bed40b1e8e7fc89cd3c0eaa157be9))_:

**Option 1**: Steps as a script:
<details>
<summary>click here for script 👩‍💻</summary>

```shell
#!/bin/sh

set -o errexit
set -o nounset

# Create directory
mkdir example && cd example

# Initialize Git repository
git init && git commit --allow-empty -m "init commit"

# Create bare bones `yarn` project with `np`
yarn init -y && yarn add -D np

# Commit everything (for simplicity)
git add --all && git commit -m "all"

# Run `np` and select `Other (specify)` for the version
yarn np --any-branch --no-tests --no-cleanup --no-release-draft --no-publish
```
</details>

**Option 2**: Steps as a individual steps:
<details>
<summary>Click here for steps 👨‍💻</summary>

1. Create directory:

    ```shell
    mkdir try && cd try
    ```

1. Initialize Git repository:

    ```shell
    git init && git commit --allow-empty -m "init commit"
    ```

1. Create bare bones `yarn` project:

    ```shell
    yarn init -y
    ````

1. Install `np`:

    ```shell
    yarn add -D np
    ```

1. Commit everything (for simplicity):

    ```shell
    git add --all && git commit -m "all"
    ````

1. Run `np` and select `Other (specify)` for the version:

    ```shell
    yarn np --any-branch --no-tests --no-cleanup --no-release-draft --no-publish
    ```

</details>

The output of the failing command looks approximately like this:

<details>
<summary>click here for output</summary>

```shell
? Select semver increment or specify new version Other (specify)

  ✔ Git
  ✖ Bumping version using Yarn
    → info Visit https://yarnpkg.com/en/docs/cli/version for documentation about this command.
    Pushing tags

✖ Command failed with exit code 1: yarn version --new-version
error Invalid version supplied.
info Current version: 1.0.0
info Visit https://yarnpkg.com/en/docs/cli/version for documentation about this command.
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

</details>

## Expected behavior

The behavior I would expect from reading the code in [ui.js](https://github.com/sindresorhus/np/blob/f72c30e65fb40822ee1b1191c227816b3334a374/source/ui.js) would be to show a second input to type in the custom version number after selecting `Other (specify)` in the interactive UI.

<!--- Tell us what you expected to happen -->

## Environment

<!-- Mention which versions of np, Node.js, npm and Git you're using, as well as your OS and version -->

np - 6.2.3
Node.js - 14.2.0
npm - 6.14.4
Git - 2.26.2
OS - macOS 10.15.4

--- a/source/ui.js
+++ b/source/ui.js
@@ -85,7 +85,7 @@ module.exports = async (options, pkg) => {
 		},
 		{
 			type: 'input',
-			name: 'version',
+			name: 'customVersion',
 			message: 'Version',
 			when: answers => !answers.version,
 			filter: input => version.isValidInput(input) ? version(pkg.version).getNewVersionFrom(input) : input,
@@ -105,7 +105,7 @@ module.exports = async (options, pkg) => {
 			type: 'list',
 			name: 'tag',
 			message: 'How should this pre-release version be tagged in npm?',
-			when: answers => options.runPublish && version.isPrereleaseOrIncrement(answers.version) && !options.tag,
+			when: answers => options.runPublish && (version.isPrereleaseOrIncrement(answers.customVersion) || version.isPrereleaseOrIncrement(answers.version)) && !options.tag,
 			choices: async () => {
 				const existingPrereleaseTags = await prereleaseTags(pkg.name);
 
@@ -121,9 +121,9 @@ module.exports = async (options, pkg) => {
 		},
 		{
 			type: 'input',
-			name: 'tag',
+			name: 'customtag',
 			message: 'Tag',
-			when: answers => options.runPublish && version.isPrereleaseOrIncrement(answers.version) && !options.tag && !answers.tag,
+			when: answers => options.runPublish && (version.isPrereleaseOrIncrement(answers.customVersion) || version.isPrereleaseOrIncrement(answers.version)) && !options.tag && !answers.tag,
 			validate: input => {
 				if (input.length === 0) {
 					return 'Please specify a tag, for example, `next`.';
@@ -193,7 +193,8 @@ module.exports = async (options, pkg) => {
 
 	return {
 		...options,
-		...answers,
+		version: answers.version || answers.customVersion,
+		tag: answers.tag || answers.customTag,
 		confirm: true,
 		repoUrl,
 		releaseNotes

--- a/source/ui.js
+++ b/source/ui.js
@@ -121,7 +121,7 @@ module.exports = async (options, pkg) => {
 		},
 		{
 			type: 'input',
-			name: 'customtag',
+			name: 'customTag',
 			message: 'Tag',
 			when: answers => options.runPublish && (version.isPrereleaseOrIncrement(answers.customVersion) || version.isPrereleaseOrIncrement(answers.version)) && !options.tag && !answers.tag,
 			validate: input => {

